src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/Utilities/ParsedDocument.cs

```diff
@@ -55,4 +58,33 @@ public ParsedDocument WithChangedRoot(SyntaxNode root, CancellationToken cancell
         var text = root.SyntaxTree.GetText(cancellationToken);
         return new ParsedDocument(Id, text, root, HostLanguageServices);
     }
+
+    public ParsedDocument WithChange(TextChange change, CancellationToken cancellationToken)
+        => WithChangedText(Text.WithChanges(change), cancellationToken);
+
+    public ParsedDocument WithChanges(IEnumerable<TextChange> changes, CancellationToken cancellationToken)
+        => WithChangedText(Text.WithChanges(changes), cancellationToken);
+
+    /// <summary>
+    /// Equivalent semantics to <see cref="Document.GetTextChangesAsync(Document, CancellationToken)"/>
+    /// </summary>
+    public IEnumerable<TextChange> GetChanges(in ParsedDocument oldDocument)
```

Comment:
could this be an IReadOnlyList as the result instead of IEnumerable?

Reply:
Can't since SourceText.GetChanges returns IList (sadly) and that does not implement IReadOnlyList. Returning IList seems worse than IEnumerable as it implies mutability.